src/BalconyBotanica/Core/Algorithm/Recommender.cs

```diff
@@ -19,42 +19,48 @@ public RecommendedPlants Recommend(RecommendedPlants insertedArray, QuizAnswers
 
             PlantData[] arrayToFilter = insertedArray.Plants;
 
-            // DONE: check sunlight and delete all the plants who can’t function with that climate
-            // TODO: If answer has "full sun" delete "full shade".
-            // TODO: if awnser had "full shade" delete "full sun".
-            if (quizAnswers.sunlight == Sunlight.FULL_SUN)
+            if (quizAnswers.sunlight > Sunlight.FILTERED_SHADE)
             {
-                Sunlight optionToRemove = Sunlight.FULL_SHADE;
                 arrayToFilter = arrayToFilter
-                                    .Where(x => x.Sunlight.Contains(optionToRemove))
-                                    .ToArray();
+                    .Where(x =>
+                        x.Sunlight.All(s =>
+                            s > Sunlight.FILTERED_SHADE))
+                    .ToArray();
             }
-            // check watering
-            // DONE: if "minimum" than delete "frequent", 
-            //      order "minimum" as first, en "avg" as second.
-            // TODO: if average than nothing, 
-            //      order "frequent" last
-            // TODO: if frequent than nothing, 
-            //      order nothing
-            if (quizAnswers.wateringSchedule == WateringSchedule.MINIMUM)
+            // TODO: the above seems to work, this part doesnt... debug with tests
```

Comment:
I think the reason this code isn't doing what is expected is because `Sunlight.FILTERED_SHADE` is somehow being treated as a special value but it is not clear to me why because it is one of the middle enum values. 

I'll try to provide the answer below because we are reaching the end of the mentorship but if you still want to figure this out for yourself, then don't read further :-)


After line 20:

```
arrayToFilter = 
    arrayToFilter.Where(plant => 
        quizAnswers.sunlight <= plant.Sunlight
    ).ToArray();
```

This is because you have the following defined enum for Sunlight:

```
public enum Sunlight
     {
         FULL_SHADE = 10,
         FILTERED_SHADE = 20,
         PART_SHADE = 30,
         FULL_SUN = 40,
     }
```

Imagine we have a quiz answer which says I have a balcony with `PART_SHADE` (value 30), then the plants we are looking for are basically all the ones with shade values 30 or less. That translates to this in the `Where` clause:

`quizAnswers.sunlight <= plant.Sunlight`

Notice how no specific `Sunlight` enum is in the code. This means the algorithm will work no matter what values of `Sunlight` you define now or in the future as long as it stays as a list of enums. This makes the code generic and future-proof.

For the rest of the algorithm, follow the same pattern for watering schedule etc